topology/graph/processor.go

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2018 Orange
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package graph
+
+import (
+	"github.com/skydive-project/skydive/common"
+)
+
+// NodeAction is a callback to perform on a node. The action is kept
+// active as long as it returns true.
+type NodeAction interface {
+	ProcessNode(g *Graph, n *Node) bool
+}
+
+// deferred represents a node action with additional info if needed
+// for cancellation.
+type deferred struct {
+	action NodeAction
+}
+
+// Processor encapsulates an indexer that will process NodeActions
+// on the nodes that filter
+type Processor struct {
+	common.RWMutex
+	*MetadataIndexer
```

Comment:
As MetadataIndexer already implement the `EventListener` implement, it would be clearer if MetadataIndexer was not embeded but be a regular member of the Processor struct.
This way, you can embed the `DefaultGraphListener` structure so you don't have to define empty `OnNodeUpdated` and `OnNodeDeleted` functions.

Reply:
yes nicer.